rpc/src/v1/helpers/light_fetch.rs

```diff
@@ -321,9 +321,15 @@ impl LightFetch {
 			BlockId::Number(x) => Some(x),
 		};
 
-		match (block_number(filter.to_block), block_number(filter.from_block)) {
-			(Some(to), Some(from)) if to < from => return Either::A(future::ok(Vec::new())),
-			(Some(_), Some(_)) => {},
+		let (from_block_number, from_block_header) = match self.client.block_header(filter.from_block) {
+			Some(from) => (from.number(), from),
+			None => return Either::A(future::err(errors::unknown_block())),
+		};
+
+		match block_number(filter.to_block) {
+			Some(to) if to < from_block_number || from_block_number > best_number 
+				=> return Either::A(future::ok(Vec::new())),
```

Comment:
Minor nit but I would put the `=>` in the previous line, I don't know what's our style on this though.